### ISSUE TITLE
Evaluate Proposal for a Power BI Embedded implementation (again)

### ISSUE THREAD

**Issue by jesusitd** [intakedesk/PowerBI-General#582]
REFLOATED !!!

- [ ] Lay out a plan for implementation
- [x] Research about the details of licensing
- [x] Research architecture of the solution
- [x] Proposed End to End testing with a Dev Instance along with EG

**Comment by jesusitd** [comment]
Research:

Tutorial: Embed Power BI content using a sample embed for your customers application
https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-sample-for-customers?tabs=net-framework
![image](https://user-images.githubusercontent.com/50643267/104514843-c7cd8880-55a6-11eb-9786-6a5e5634336a.png)

Row-level security with Power BI Embedded | Applying user and role to an embed token
https://docs.microsoft.com/en-us/power-bi/developer/embedded/embedded-row-level-security#applying-user-and-role-to-an-embed-token

Power BI Embedded for PHP using CURL (Service Principal / App owns data scheme)
https://www.dataislight.com/2018/05/21/power-bi-embedded-with-php-curl-app-owns-data/

**Comment by jesusitd** [comment]
First thoughts:

Pricing:
![image](https://user-images.githubusercontent.com/50643267/104517481-ecc3fa80-55aa-11eb-8c4d-568dc91d1afc.png)

Auth mode: SERVICE PRINCIPAL (Application owns the data). **Must develop in CRM side.**

Row-level security (RLS): YES, by implementing Roles in Power BI (on my side). Later on, credentials for "Effective User" should be passed from the application to Power BI to render the correct data. **Must test this**.

Steps:
- Roll out a PBI Embedded instance. (RAY)
- Create a Power BI Application CLIENT+SECRET pair. (RAY)
- Develop the CRM side of application authentication. (EG)
- Create a Power BI report with some roles and users. (JG)
- TEST RLS and embedding by passing Effective Credentials from the CRM.

@fedegarza IMPORTANT please read.

**Comment by fedegarza** [comment]
- Shared Dataset?

**Comment by jesusitd** [comment]
Things to test:
- [x] Embedding with Application auth
- [x] Passing effective identifies
- [x] RLS with Power BI
- [x] Shared datasets

**Comment by jesusitd** [comment]
> * Shared Dataset?

Yes. It also works with RLS on Shared dataset using fake Effective Identities
![image](https://user-images.githubusercontent.com/50643267/124972463-6556c180-dff8-11eb-8c38-79f932b998f5.png)

CC @fedegarza
